src/robo_jogador.py

```
@@ -21,77 +45,173 @@ def sense_act(self, stdscr):
-                elif isinstance(cell, str) and cell not in ['#']:
-                    stdscr.addstr(20, 0, f"Duelando contra {cell} em {target}...\n")
-                    self.duelo(cell, target)
+
+
+                elif poder_inimigo > poder_atacante:
+                    # Inimigo vence
+                    self.logger.info(f"{inimigo_id} venceu o duelo contra {self.id}!")
+                    stdscr.addstr(20, 0, f"{inimigo_id} venceu o duelo contra {self.id}!\\n")
+                    # Marcar atacante como morto e liberar sua célula
+                    self._marcar_morto(self.id, self.pos)
+                    selfscr.addstr(20, 0, f"Você morreu em combate! Fim de jogo.\\n")
```

Ruff (0.11.9)

Undefined name `selfscr`

(F821)